src/Nancy/Routing/DefaultRouteInvoker.cs

```diff
@@ -35,7 +33,7 @@ public DefaultRouteInvoker(IResponseNegotiator negotiator)
         /// <returns>A <see cref="Response"/> instance that represents the result of the invoked route.</returns>
         public async Task<Response> Invoke(Route route, CancellationToken cancellationToken, DynamicDictionary parameters, NancyContext context)
         {
-            dynamic result;
+            object result;

```

Comment:
Because outside of the actual route lambdas, we don't need `dynamic` .. using `dynamic` in other places has side effects such as the entire method body being treated as dynamic so the compiler cannot infer a return type tc